src/client/datascience/data-viewing/data-wrangler/dataWranglerJupyterVariableDataProvider.ts

```diff
@@ -0,0 +1,188 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+'use strict';
+import '../../../common/extensions';
+
+import { inject, injectable, named } from 'inversify';
+
+import { Identifiers } from '../../constants';
+import { IJupyterVariable, IJupyterVariables, INotebook } from '../../types';
+import { DataViewerDependencyService } from '../dataViewerDependencyService';
+import { ColumnType, IDataFrameInfo, IRowsResponse, IColsResponse } from '../types';
+import { traceError } from '../../../common/logger';
+import { IDataWranglerJupyterVariableDataProvider } from './types';
+
+@injectable()
+export class DataWranglerJupyterVariableDataProvider implements IDataWranglerJupyterVariableDataProvider {
```

Comment:
Is this much different than IJupyterVariables? Why not add the new methods to IJupyterVariables and just use those? 

That would have the added benefit of supporting doing this during debugging too (although that's not a likely use case)

Reply:
I guess I was scared of adding more methods to the interface in case other components relied on them and would have to update them, but I see your point and will just add it there.